### ISSUE TITLE
Section about specifying INSTANCENAME

### ISSUE THREAD

**Issue by andy-maier** [pywbem/pywbemtools#282]
Issuing a command that needs an INSTANCENAME fails as follows when specifying the syntax shown in the usage examples.

```
$ pywbemcli -m tests/unit/simple_mock_model.mof instance get CIM_Foo.InstanceID=CIM_Foo1 -n root/cimv2
Error: Invalid wbem uri input :CIM_Foo.InstanceID=CIM_Foo1. Error WBEM URI has invalid value format in a keybinding: 'InstanceID'='CIM_Foo1'
```

Specifying the keybinding value with double quotes does not help:

```
$ pywbemcli -m tests/unit/simple_mock_model.mof instance get CIM_Foo.InstanceID="CIM_Foo1" -n root/cimv2
Error: Invalid wbem uri input :CIM_Foo.InstanceID=CIM_Foo1. Error WBEM URI has invalid value format in a keybinding: 'InstanceID'='CIM_Foo1'
```

Some more information, showing that the instance exists and the specified property is a key property:

```
$ pywbemcli -m tests/unit/simple_mock_model.mof -o table instance enumerate CIM_Foo
Instances: CIM_Foo
+--------------+---------------+
| InstanceID   | IntegerProp   |
|--------------+---------------|
| "CIM_Foo1"   | 1             |
| "CIM_Foo2"   | 2             |
| "CIM_Foo3"   |               |
+--------------+---------------+

$ pywbemcli -m tests/unit/simple_mock_model.mof class get CIM_Foo -n root/cimv2
   [Description ( "Simple CIM Class" )]
class CIM_Foo {

      [Key ( true ),
       Description ( "This is key property." )]
   string InstanceID;

      [Description ( "This is Uint32 property." )]
   uint32 IntegerProp;

      [Description ( "Method with in and out parameters" )]
   uint32 Fuzzy(
         [IN ( true ),
          OUT ( true ),
          Description ( "Define data to be returned in output parameter" )]
      string TestInOutParameter,
         [IN ( true ),
          OUT ( true ),
          Description ( "Test of ref in/out parameter" )]
      CIM_Foo REF TestRef,
         [IN ( false ),
          OUT ( true ),
          Description ( "Rtns method name if exists on input" )]
      string OutputParam,
         [IN ( true ),
          Description ( "Defines return value if provided." )]
      uint32 OutputRtnValue);

      [Description ( "Method with no Parameters" )]
   uint32 DeleteNothing();

};
```

**Comment by andy-maier** [comment]
This is so error prone that I think we need to describe this in its own section, e.g. as a new subsection of 4.1. "Pywbemcli special command line features". In addition, I like your suggestion to improve the exception message.

Note: In PR #295, I removed the ABNF for WBEM-URI because I think that people can look it up in DSP0207 which we link to, but it did not address the shell escaping, nor should it in this glossary-type text.

**Comment by andy-maier** [comment]
Will be addressed when describing how to specify an INSTANCENAME as suggested in PR #295.

**Comment by andy-maier** [comment]
Here is the proposal from PR #295 for defining INSTANCENAME:

If we have any ABNF anywhere, i suggest that we reduce it to just what is needed for INSTANCENAME. Because that is the only place we need it in pywbemcli. Then we can restrict it to instances only, to untyped only, and can omit the host part.

That leaves something like this:

```
INSTANCENAME = CLASSNAME "." keybindings
keybindings = keybinding *( "," keybinding )
keybinding = PROPERTYNAME "=" value
value = constantValue / "\"" escaped_instancename "\""
```
where `constantValue` is defined in DSP0004, and `escaped_instancename` is a backslash-escaped `INSTANCENAME`.

**Comment by andy-maier** [comment]
This issue has become a docs issue for writing a section on how to specify INSTANCENAME. It needs to wait until PR #290 is merged and use that as a basis.

**Comment by andy-maier** [comment]
Addressed by PR #303.
